Code/Framework/AzFramework/AzFramework/Input/Devices/Gamepad/InputDeviceGamepad.h

```diff
@@ -182,6 +182,14 @@ namespace AzFramework
         // Reflection
         static void Reflect(AZ::ReflectContext* context);
 
+        ////////////////////////////////////////////////////////////////////////////////////////////
+        // Foward declare the internal Implementation class so it can be passed into the constructor
+        class Implementation;
+
+        ////////////////////////////////////////////////////////////////////////////////////////////
+        //! Alias for the function type used to create a custom implementation for this input device
+        using ImplementationFactory = typename InputDeviceGamepad::Implementation*(InputDeviceGamepad&);
```

Comment:
FYI the `typename InputDeviceGamepad` part of the type alais isn't needed.

```suggestion
        using ImplementationFactory = Implementation*(InputDeviceGamepad&);
```

Reply:
Good call - I originally had:

`using ImplementationFactory = typename Implementation*(InputDeviceGamepad&);`

but that broke clang so I just added InputDeviceGamepad:: - your suggestion without 'typename' works though, thanks!

I also updated the InputDeviceImplementationRequest bus to use the same ImplementationFactory type alias.